src/test/java/com/google/sps/servlets/UserFoldersServletTest.java

```diff
@@ -0,0 +1,180 @@
+package com.google.sps.servlets;
+
+import static com.google.sps.tool.Tool.compareJson;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.After;
+
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+
+import com.google.sps.data.Folder;
+import java.util.List; 
+import java.util.ArrayList;
+
+@RunWith(JUnit4.class)
+public final class UserFoldersServletTest {
+
+  private static final String USER_KIND = "User";
+  private static final String USER_ID = "testId";
+  private static final String FOLDER_A = "First Folder";
+  private static final String FOLDER_B = "Second Folder";
+  private static final String FOLDER_LANGUAGE = "en";
```

Comment:
These are small enough that they should be hardcoded into each test.

Reply:
Also, their names don't give the test reader enough context to understand what they contain, so they also don't follow the rules I posted in the chat about testing constants.